### PR TITLE
Add np.fft.fftshift/ifftshift

### PR DESCRIPTION
**Pull request by adler-j**
Ported fftshift functions from numpy

### CODE REVIEW

**Comment by hawkinsp** [comment]
The code looks fine, but I'd suggest adding a test to `fft_test.py`.

**Comment by mattjj** [comment]
Ping @adler-j : are you up for adding a test? (If not we might be able to open an issue for it, and mark it as a "good first issue" for someone to jump in and contribute.)

**Comment by adler-j** [comment]
I'll try to get some time to add a test in the coming days!

**Comment by adler-j** [comment]
Added the tests as requested.

**Comment by adler-j** [comment]
Test failure seems to be due to flakyness, how do I deal with that?

File: tests/fft_test.py [hunk 1]
```diff
@@ -353,5 +353,51 @@ def testRfftfreqErrors(self, n):
...
+    np_fn = lambda arg: np_op(arg)
+    onp_fn = lambda arg: onp_op(arg)
+    # Numpy promotes to complex128 aggressively.
+    self._CheckAgainstNumpy(onp_fn, np_fn, args_maker, check_dtypes=False,
+                            tol=1e-4)
+    self._CompileAndCheck(np_fn, args_maker, check_dtypes=True)
+    # Test gradient for differentiable types.
+    if dtype in inexact_dtypes:
+      tol = 0.15  # TODO(skye): can we be more precise?
+      jtu.check_grads(np_fn, args_maker(), order=2, atol=tol, rtol=tol)
```
Comment (shoyer) [inline tests/fft_test.py]: I think this can all be replaced by just:
```python
    self._CheckAgainstNumpy(onp_fn, np_fn, args_maker, check_dtypes=True)
```

The comment on complex128 is not relevant to fftshift/iffshift. And we not really need to check compiling or gradients here, since these functions are built-up out of the independently tested `jax.numpy.roll`.

**Comment by shoyer** [comment]
> Test failure seems to be due to flakyness, how do I deal with that?

I tried restarting it but it failed again. Can you try merging in the latest changes from master into your branch? (My guess is that the flake as has been fixed.)

File: tests/fft_test.py (hunk 1 above)
Comment (adler-j) [reply tests/fft_test.py]: Nice comment, updated according to this (although there might be more culprits with the same problem laying around in this file, that's for later)

**Comment by adler-j** [comment]
Rebased on master, lets see if it's fixed.

File: tests/fft_test.py [hunk 2]
```diff
@@ -353,5 +353,33 @@ def testRfftfreqErrors(self, n):
       lambda: func(n=10, d=n)
     )
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_shape={}".format(
+      jtu.format_shape_dtype_string(shape, dtype)),
+      "dtype": dtype, "shape": shape, "rng_factory": rng_factory}
+    for rng_factory in [jtu.rand_default]
+    for dtype in all_dtypes
+    for shape in [[9], [10], [101], [102], [3, 5], [3, 17], [5, 7, 11]]))
```
Comment (shoyer) [inline tests/fft_test.py]: Looking over these tests, I noticed that there does not seems to be any test coverage for providing the `axes` argument. Could you please add that?

**Comment by hawkinsp** [comment]
It looks like there's a travis failure in one of the FFT tests.

**Comment by adler-j** [comment]
Added test for axes and fixed travis failure, ready?

**Comment by shoyer** [comment]
Yes, thank you!
